core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -497,13 +502,41 @@ public boolean isMailerPluginPresent() {
 
     /**
      * Creates a new user account by registering a password to the user.
+     *
+     * Supports bcrypt hashed passwords if the password begins with the characters
+     * <code>#jbcrypt:</code>
      */
     public User createAccount(String userName, String password) throws IOException {
         User user = User.getById(userName, true);
         user.addProperty(Details.fromPlainPassword(password));
         return user;
     }
 
+    /**
+     * Creates a new user account by registering a JBCrypt Hashed password with the user.
+     *
+     * @param userName The user's name
+     * @param hashedPassword A hashed password, must begin with <code>#jbcrypt:</code>
+     */
+    public User createAccountWithHashedPassword(String userName, String hashedPassword) throws IOException {
+        if (!hashedPassword.startsWith(JBCRYPT_HEADER)) {
```

Comment:
I don't like how this method has to use the duplicated string. I'm not sure the check is needed but at a minimum it should probably use the `isPasswordHashed()` method.

Reply:
The check definitely is necessary, otherwise you would be able to supply a non-hashed password to this method, which would violate the assumptions made later down.